api/http-api.md

```diff
@@ -0,0 +1,176 @@
+  * `RegisterNamespace` - `POST /api/v1/namespaces`
+    * 202 on success since eventually consistent
+  * `DescribeNamespace` - `GET /api/v1/namespaces/{namespace}`
+  * `ListNamespaces` - `GET /api/v1/namespaces`
+  * `UpdateNamespace` - `PUT /api/v1/namespaces/{namespace}`
+    * TODO(cretz): Confirm strongly consistent and idempotent or not
+  * `DeprecateNamespace` - `DELETE /api/v1/namespaces/{namespace}`
+    * Is this eventually consistent? Will determine response code
+    * Does this prevent returning from `GET`? Will determine HTTP method
+* Workflows:
+  * `StartWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}`
+    * 204 on success
+    * Should accept header as `identity` on these calls?
+    * Ok to be at root URL here instead of appending `/start`?
+  * `GetWorkflowExecutionHistory` - `GET /api/v1/namespaces/{namespace}/workflows/{workflow_id}/history`
+    * Accept `?nextPageToken` as query param? I'm thinking so
+  * `GetWorkflowExecutionHistoryReverse` - `GET /api/v1/namespaces/{namespace}/workflows/{workflow_id}/history-reverse`
+  * `RequestCancelWorkflowExecutionRequest` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/cancel`
+    * Intentionally not taking the run ID in the URL, but should it be `?runId=` query param?
+  * `SignalWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/signal`
+    * Should we put signal name in the URL? Can be optional (e.g. in URL or body)
+  * `SignalWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/signal-with-start`
+  * `ResetWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/reset`
+  * `TerminateWorkflowExecution` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/terminate`
+  * `ListOpenWorkflowExecutions` - Intentionally not exposed
+  * `ListClosedWorkflowExecutions` - Intentionally not exposed
+  * `ListWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/workflows`
+    * Accepts `?query=` query parameter
+  * `ListArchivedWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/archived-workflows`
+    * Intentionally did not do `/workflows/archived` due to ambiguity
+  * `CountWorkflowExecutions` - `GET /api/v1/namespaces/{namespace}/workflow-count`
+  * `QueryWorkflow` - `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/query`
+    * Should we put query name in the URL? Can be optional (e.g. in URL or body)
+  * `DescribeWorkflowExecution` - `GET /api/v1/namespaces/{namespace}/workflows/{workflow_id}`
+  * `UpdateWorkflowExecution`- `POST /api/v1/namespaces/{namespace}/workflows/{workflow_id}/updates/{update_id}`
+    * Decided to make this an identity so we can have "describe update" or "cancel update" or whatever one day
+  * `PollWorkflowExecutionUpdate` - Intentionally not exposed
+    * Not sure if best to have long HTTP API here. Would prefer non-blocking describe and maybe one day the difference
```

Comment:
I vote for polling, it's a useful enough API and we'll definitely want that in our UI.

Reply:
This needs to shake out better. It's too young. Until we have a non-blocking poll-update call I don't think it's ready for HTTP-ification.

Reply:
Can wait, but I would definitely want this in the HTTP API eventually.

Reply:
I vote no for polling. Keep it minimal until we see enough interest with valid use cases.

Reply:
:+1: It is so easy to add that @bergundy can alter a few lines here during the development of the thing that needs it and we can merge it in then.

Reply:
Added poll calls as non-goal currently